lib/services/livesync/platform-livesync-service-base.ts

```diff
@@ -78,7 +79,7 @@ export abstract class PlatformLiveSyncServiceBase implements IPlatformLiveSyncSe
 			device.applicationManager.checkForApplicationUpdates().wait();
 
 			let appIdentifier = this.liveSyncData.appIdentifier;
-			if (!device.applicationManager.isApplicationInstalled(appIdentifier).wait()) {
+			if (!device.applicationManager.isApplicationInstalled(appIdentifier).wait() || !this.$platformService.isPlatformBuilt(deviceAppData.platform.toLowerCase()).wait()) {
```

Comment:
We need this because in some cases the build folder is removed (for example the prepareProject cleans the build folder on android when there is a change in tns-core-modules, or the user simply removes the folder). In this situation there is no apk file and we cannot guarantee that the application installed on device is the same application that we want. I think the best option in this scenario is to build the application again. This way we will ensure that the correct application is installed.